balancer.go

```diff
@@ -54,6 +55,35 @@ func (rr *RoundRobin) balance(partitions []int) int {
 	return partitions[offset%length]
 }
 
+// ChunkedRoundRobin is a Balancer implementation that equally distributes messages
+// across all available partitions, but puts greater emphasis on batching by a chunk size
+// within a shorter time period than is possible via the regular RoundRobin Balancer.
+type ChunkedRoundRobin struct {
+	chunkSize int
+
+	mutex   sync.RWMutex
+	counter *uint64
+}
+
+// Balance satisfies the Balancer interface.
+func (rr *ChunkedRoundRobin) Balance(msg Message, partitions ...int) int {
+	rr.mutex.RLock()
+	if rr.counter == nil {
+		var nextToInsert uint64 = math.MaxUint64 // so that first increment overflows it back to 0
```

Comment:
We need to lock the mutex for writes within this if statement

Reply:
I think it might make more sense to just use `rr.mutex.Lock()` above and `rr.mutex.Unlock()` below

Reply:
Yeah, this is where I was thinking copying bulrush was a mistake.  Does it make sense to do the same locking as LeastBytes?
`
lb.mutex.Lock()

defer lb.mutex.Unlock()
`

Reply:
Yeah adding the defer in for unlocks is probably the safest option. It will add some additional locking time but that won't be meaningful